modules/images/webp-uploads/load.php

```diff
@@ -340,3 +340,130 @@ function webp_uploads_wp_get_missing_image_subsizes( $missing_sizes, $image_meta
 }
 
 add_filter( 'wp_get_missing_image_subsizes', 'webp_uploads_wp_get_missing_image_subsizes', 10, 3 );
+
+/**
+ * Filters `the_content` to update images so that they use the preferred MIME type where possible.
+ *
+ * By default, this is `image/webp`, if the current attachment contains the targeted MIME
+ * type. In the near future this will be filterable.
+ *
+ * Note that most of this function will not be needed for an eventual core implementation as it
+ * would rely on `wp_filter_content_tags()`.
+ *
+ * @since n.e.x.t
+ *
+ * @see wp_filter_content_tags()
+ *
+ * @param string $content The content of the current post.
+ * @return string The content with the updated references to the images.
+ */
+function webp_uploads_update_image_references( $content ) {
+	// This content does not have any tag on it, move forward.
+	if ( ! preg_match_all( '/<(img)\s[^>]+>/', $content, $img_tags, PREG_SET_ORDER ) ) {
+		return $content;
+	}
+
+	$images = array();
+	foreach ( $img_tags as list( $img ) ) {
+		// Find the ID of each image by the class.
+		if ( ! preg_match( '/wp-image-([\d]+)/i', $img, $class_name ) ) {
+			continue;
+		}
+
+		if ( empty( $class_name ) ) {
+			continue;
+		}
+
+		// Make sure we use the last item on the list of matches.
+		$attachment_id = (int) end( $class_name );
```

Comment:
Nit-pick, but IMO safer to not use an iterator:
```suggestion
		$attachment_id = (int) $class_name[1];
```